src/decoder/lattice-faster-decoder.cc

```diff
@@ -683,6 +684,7 @@ BaseFloat LatticeFasterDecoder::GetCutoff(Elem *list_head, size_t *tok_count,
   }
 }
 
+template <typename Fst>
```

Comment:
Please call this FstType instead of Fst: otherwise the code is hard to read (confusion with fst::Fst).